client/signup/main.jsx

```diff
@@ -77,6 +77,15 @@ class Signup extends React.Component {
 		store: PropTypes.object,
 	};
 
+	static propTypes = {
+		domainsWithPlansOnly: PropTypes.bool,
+		isLoggedIn: PropTypes.bool,
+		loadTrackingTool: PropTypes.func.isRequired,
+		setSurvey: PropTypes.func.isRequired,
+		signupDependencies: PropTypes.object,
+		trackAffiliateReferral: PropTypes.func.isRequired,
+	};
+
 	constructor( props, context ) {
 		super( props, context );
 		SignupDependencyStore.setReduxStore( context.store );
```

Comment:
An idea for another cleanup: for Flux/Redux interaction, we use `lib/redux-bridge` and its `reduxGetState` and `reduxDispatch` exports. Disclaimer: I'm not sure it really makes sense for signup, which has standalone Redux store different from the rest of Calypso.

Reply:
Good to know - I'd been wondering what `redux-bridge` was. 

Sign up has a lot of tech debt besides being written in Flux, so just slapping on `redux-bridge` might not be sufficient. 

I'd gone ahead and given this part of the codebase some tender loving care the past few days, so keep a lookout for future refactor/reduxification PRs :)